src/Discord.Net.Commands/Extensions/CommandServiceExtensions.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Discord.Commands
+{
+    public static class CommandServiceExtensions
+    {
+        public static async Task<IReadOnlyCollection<CommandInfo>> GetExecutableCommandsAsync(this IEnumerable<CommandInfo> commands, ICommandContext context, IServiceProvider provider)
```

Comment:
Now that I think about it, it's probably not a good idea to just allow `IEnumerable` for this. Maybe we should accept `ICollection<CommandInfo>` instead, since that's what we return.

Reply:
The `Commands` property on the service is exposed as an `IEnumerable`, so this change would make `GetExecutableCommandsAsync(commandService.Commands)` impossible without a type change/casting.